cl/favorites/utils.py

```diff
@@ -200,30 +252,76 @@ def send_prayer_emails(instance: RECAPDocument) -> None:
 
 
 async def get_user_prayer_history(user: User) -> tuple[int, float]:
-    filtered_list = Prayer.objects.filter(user=user, status=Prayer.GRANTED)
+    filtered_list = Prayer.objects.filter(
+        user=user, status=Prayer.GRANTED
+    ).select_related("recap_document")
 
     count = await filtered_list.acount()
 
     total_cost = 0
     async for prayer in filtered_list:
```

Comment:
I'm concerned about the performance of this `total_cost` computation if the user has thousands of prayers.

Would it be possible to use a similar approach, relying on a database query like the one you used in `get_lifetime_prayer_stats` to compute the cost?

Reply:
> Would it be possible to use a similar approach, relying on a database query like the one you used in get_lifetime_prayer_stats to compute the cost?

Yes, that's possible. I'll update this helper function accordingly.